CMakeLists.txt

```diff
@@ -43,3 +43,4 @@ add_subdirectory(test)
 
 # add usage example
 add_subdirectory(example)
+add_subdirectory(example/third_party_examples)
```

Comment:
I will merge the examples as they are but please do not feel offended when I refactor them a bit (clang-format, removal of `this->`, removal of `const` in data members to make the class a regular type, etc). As I  wrote in general comments I will leave custom namespaces (I know that is your preferred syntax 😉) to have some time to get used to it and after that, we will decide which one should be advertised by the library documentation and examples.

Reply:
I would wait a while. I am abusing this pull request to check build atm since local build isnt working. I will pull this and reformat it in a prettier way and do another pull request once I have examples done. How about that, so comments and issues welcomed.. then can do a clean pull request

Reply:
Hmm, I am really not sure what problem do you have with fmt. Could you please ping me on Cppslack ('mpusz') and we will try to solve it?

Reply:
On the length::mm style syntax, obviously I like it. I am not trying to force it ( so then "third_party" is the right characteristic), but it has all the information , is terse for the quantities I tend to use, just seems to work well in practise, and have done it that way for a long time, so feels natural to me.  With more modern c++ techniques I think that length::mm<rep = system_float_type> is best. The one part you want to change is the numeric type underneath. I use float , rather than double as default for [STM32F4](https://www.st.com/en/microcontrollers-microprocessors/stm32f4-series.html) since they have a single precision hardware floating point unit. Int is useful for transport since it is predictable, etc. ( Int  also would be interesting for e.g [arduino](https://www.arduino.cc/) users with very constrained hardware, as long as its semantics are predictable in conversions etc))